### ISSUE TITLE
brew install netcdf --with-fortran fails.

### ISSUE THREAD

**Issue by spatala** [Homebrew/homebrew-science#5420]
### Indicate troubleshooting steps taken

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Checked that a similar issue hasn't already been filed?
- [ ] Looked up the software bug tracker if reporting a build error?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?
No permission errors.

### Bug reports / build errors:
==> Installing netcdf from homebrew/science
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Downloading ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-4.4.1.1.tar.gz
Already downloaded: /Users/patala/Library/Caches/Homebrew/netcdf-4.4.1.1.tar.gz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/netcdf/4.4.1.1_4 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_SHARED_LIBS=ON -DNC_EXTRA_DEPS=-lmpi -DENABLE_DAP_AUTH_TESTS=OFF -DENABLE_NETCDF_4=ON -DENABLE_DOXYGEN=O
==> make
==> make test
Last 15 lines from /Users/patala/Library/Logs/Homebrew/netcdf/03.make:
	163 - ncdap_tst_formatx (Failed)
	164 - ncdap_test_partvar (OTHER_FAULT)
	166 - ncdap_test_nstride_cached (OTHER_FAULT)
	167 - ncdap_t_misc (OTHER_FAULT)
	168 - ncdap_test_varm3 (OTHER_FAULT)
	169 - C_tests_simple_xy_wr (OTHER_FAULT)
	170 - C_tests_simple_xy_rd (OTHER_FAULT)
	171 - C_tests_sfc_pres_temp_wr (OTHER_FAULT)
	172 - C_tests_sfc_pres_temp_rd (OTHER_FAULT)
	173 - C_tests_pres_temp_4D_wr (OTHER_FAULT)
	174 - C_tests_pres_temp_4D_rd (OTHER_FAULT)
	175 - cdl_create_sample_files (Failed)
	176 - cdl_do_comps (Failed)
Errors while running CTest
make: *** [test] Error 8

READ THIS: http://docs.brew.sh/Troubleshooting.html

These open issues may also help:
netcdf error https://github.com/Homebrew/homebrew-science/issues/5139
netCDF and authentication/authorization https://github.com/Homebrew/homebrew-science/issues/4843
netcdf: fixes Homebrew/homebrew-science#4843 https://github.com/Homebrew/homebrew-science/pull/4847
netcdf fortran fails configure stage with brew installed gfortran https://github.com/Homebrew/homebrew-science/issues/4316
NetCDF fails to link using Homebrew-installed GCC https://github.com/Homebrew/homebrew-science/issues/3536

#### Brief summary of issue:
when I run `brew install netcdf --with-fortran`, I get `Errors while running CTest`.

#### The complete command that you issued:
`brew install netcdf --with-fortran`

#### Gist with output of `brew doctor`:
Your system is ready to brew.

#### Link output by `brew gist-logs <formula>`:
[https://gist.github.com/161e242981c362bd5689df35be53e2b3]

#### Output of `brew config`:
HOMEBREW_VERSION: 1.1.12
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 8ea778f7c7ea790694485030d105a92a9416ef33
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d9587d49b98043dfc18a2a95d658d24347b23759
Core tap last commit: 63 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.12.2 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.1_1/bin/ruby
Java: 1.8.0_92
macOS: 10.12.4-x86_64
Xcode: 8.3.1
CLT: 8.3.0.0.1.1490382677
X11: 2.7.11 => /opt/X11

### Formula Requests:

**The quickest way to get a new formula into Homebrew is to file a [Pull Request](https://github.com/Homebrew/homebrew-science/pulls)**.

We will consider your request but it may be closed if it's something we're not actively planning to work on.

**Comment by bhelenbr** [comment]
I believe this will be fixed once this pull request is merged:  https://github.com/Homebrew/homebrew-core/pull/12198.  I'm not sure when that happens though.

**Comment by bhelenbr** [comment]
If you "brew install --HEAD cmake", this problem also goes away.

**Comment by spatala** [comment]
I uninstalled cmake and then reinstalled it `brew install --HEAD cmake` and then ran `brew install netcdf --with-fortran`. **I am still getting the same error.**
Any idea why this would be the case? Thank you.

**Comment by bhelenbr** [comment]
My homebrew seems to be more up to date than yours:

HOMEBREW_VERSION: 1.1.13-14-g417f49dd
ORIGIN: https://github.com/Homebrew/brew
HEAD: 417f49dd32bfe555b20f3f7d88e2739f6623991b
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e2bf7ff3fccd4ebf7f0b3bb18d5b1577052982d7
Core tap last commit: 2 weeks ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit arrandale
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.11.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12.4-x86_64
Xcode: 8.3.1
CLT: 8.3.0.0.1.1490382677
X11: 2.7.11 => /opt/X11

**Comment by iMichka** [comment]
This has been fixed by https://github.com/Homebrew/homebrew-core/pull/12198
If this is not the case, please open a new issue and fill out the full issue template. Thanks.

**Comment by spatala** [comment]
@bhelenbr 
You are right, your homebrew version seems to be more up to date than mine. The output of `brew config` on my computer is shown below. 

I tried `brew update` and `brew upgrade`. Doesn't seem to update the version for some reason. Any tips?

HOMEBREW_VERSION: 1.1.13
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 46f6dc4c93b96abd528c3a0a808a5ca5a795c7f3
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5e6869c30a24a3a1c7aa35e9e80ef8e4c5430b13
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.12.2 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.1_1/bin/ruby
Java: 1.8.0_92
macOS: 10.12.4-x86_64
Xcode: 8.3.2
CLT: 8.3.0.0.1.1490382677
X11: 2.7.11 => /opt/X11

**Comment by bhelenbr** [comment]
Try deleting and starting over?  I’m not very sure why that is the case.

**Comment by iMichka** [comment]
It's not really about the core homebrew library, but about cmake. What does `brew info cmake` say?

**Comment by spatala** [comment]
@iMichka 
The output of `brew info cmake` is as follows:

cmake: stable 3.8.0 (bottled), HEAD
Cross-platform make
https://www.cmake.org/
/usr/local/Cellar/cmake/HEAD-3cb7048 (2,291 files, 29.5MB) *
  Built from source on 2017-04-24 at 14:14:48
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/cmake.rb
==> Dependencies
Build: sphinx-doc ✔
==> Options
--with-completion
	Install Bash completion (Has potential problems with system bash)
--without-docs
	Don't build man pages
--HEAD
	Install HEAD version
==> Caveats
Emacs Lisp files have been installed to:
  /usr/local/share/emacs/site-lisp/cmake

Actually, according to @bhelenbr 's suggestion, I uninstalled and re-installed brew. Everything seems to be working now. However, my brew config, I think, is still older than @bhelenbr 's version. Here is the output of `brew config`

HOMEBREW_VERSION: 1.1.13
ORIGIN: https://github.com/Homebrew/brew
HEAD: 46f6dc4c93b96abd528c3a0a808a5ca5a795c7f3
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: de7bac2d35d3c70524e5f4b4d27487784ee0e7cc
Core tap last commit: 60 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit ivybridge
Homebrew Ruby: 2.0.0-p648
Clang: 8.1 build 802
Git: 2.11.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_92
macOS: 10.12.4-x86_64
Xcode: 8.3.2
CLT: 8.3.2.0.1.1492020469
X11: 2.7.11 => /opt/X11

Thank you!

**Comment by iMichka** [comment]
> Actually, according to @bhelenbr 's suggestion, I uninstalled and re-installed brew. Everything seems to be working now. However, my brew config, I think, is still older than @bhelenbr 's version. Here is the output of brew config

OK so the actual netcdf error is fixed for you. If you think your homebrew does not update correctly, please report this there: https://github.com/Homebrew/brew. Thanks.

Closing here.
